File: src/plugin/combat/magic/modern/teleport/MobilisingArmiesTeleportSpell.java
package plugin.combat.magic.modern.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/27/2017
 */
public class MobilisingArmiesTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 10;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(LAW_RUNE, 1, WATER_RUNE, 1, AIR_RUNE, 1);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2413, 2848, 0);
	}
	
	@Override
	public int spellId() {
		return 37;
	}
	
	@Override
	public double exp() {
		return 19;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.REGULAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/modern/teleport/MobilisingArmiesTeleportSpell.kt
package plugin.interaction.combat.magic.modern.teleport

import game.content.plugin.combat.spell.type.TeleportSpellPlugin
import game.global.WorldTile
import utility.constants.MagicConstants
import utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/27/2017
 */
class MobilisingArmiesTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 10
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.LAW_RUNE, 1, MagicConstants.WATER_RUNE, 1, MagicConstants.AIR_RUNE, 1)
    }

    override fun destination(): WorldTile {
        return WorldTile(2413, 2848, 0)
    }

    override fun spellId(): Int {
        return 37
    }

    override fun exp(): Double {
        return 19.0
    }

    override fun book(): MagicBook {
        return MagicBook.REGULAR
    }
}